### PR TITLE
#5480 Add checks for incoming pubkey if its already in the user's contact list

### LINKED ISSUE #5480: Use 'Import key' instead of 'Update key' on the import public key screen
1. No keys for rich@flowcrypt.test image
2. Import the first key
image
3. We see `Import key`. Looks good. image 
image

4. Try to import a new one(different fingerprint). It shows `Update key` but should show `Import key` as we import a totally different key, not updating the existing key 
image
image

5. Also it would be nice to fix UI to show info about a key in a single line.

<details>
  <summary>Key 1</summary>

```
-----BEGIN PGP PUBLIC KEY BLOCK-----
Version: FlowCrypt 1.4.7_dev_147__2023_11_09__3227f2d Gmail Encryption
Comment: Seamlessly send and receive encrypted email

mQINBGU7ZP8BEAChOXoPiC2b/eDztF5J2drxv6JSWRvhtR6bowA18JM9tkJ6smUb
oINQgwth4GNnSSaWOvgPwExIXfPs/p/YY6ULbhe7i8PWyL+48ugLnuVsR0mmoRov
T+1FRWCR1RW8/50Vq2efUCCxpWRthAXVbqIisINXqU8tQQ0FT1ubVXWG99wDIR1+
V+d7+Gh3NjPhroJJtSEd48/kPp/UTUgnyyezL8AOt3FHWdQvsJl4//kJHNGtvIoM
p5rKflTHncmQlMJqmd/n9hqbGGHiK2zSEc2lLRr6kEW2CW/nJt7vgyTpBBunDOS8
VJwc6iKeYe1vCtNC2Xtqia9/rxjSm2MbNgoVprEO+W1xi6T2U5yY8yJ10llzKInj
K3Hi2r3GwInuSnygI76+uX04glvYibNxydx5fOIw/lR61fKeAWvzoFiH0lQ6hBZe
o5QSXtq4tUCg1rFA9FIn2RNuQEZoS3LM5HdhphDkXScUNTk89k6xFURRmvGzpSma
fbWYtg+kSfl7KBSkeTrtlLDo5bOFKavGfu2//pNYquN75sza8TuE3Gx0Pfcn+iV0
Jz5NfbYz+FkhZdwWIKVMY/lAGp27MY10Lq8ytrjThYd8rNyi7yb9AmGxcGHEjG8T
h0EN5Q0yvCd9nCNAvXsh9x5XSm77PxS3Fpt4OJmIyJt5H2q2NFlM6h+2rQARAQAB
tBpSaWNoIDxyaWNoQGZsb3djcnlwdC50ZXN0PokCUwQTAQgAPQKeAQKbAQUWAgMB
AAQLCQgHBRUKCQgLApkBFiEEZw9JBGKUIT6hZtumqtVVDfvHnyIFAmVKVu8FCQWy
jHAACgkQqtVVDfvHnyJO4g/9H3mSK/nxC4gobU5zav3IcVdqfl9YKYD4RjLx4Jt/
qRbs0pv2di6sKdBqmpXJP6RsXUmbkF9bAeMpioCjEUNM6WRHrJCd+Y+7UHaieymC
+WiVxAyDuWkTRwuEgsIMNQ6ybO/VR1xi2gdEA73Mg2jGUHeXAzz4yuI1byKDY/Tu
+0TtLK8q7y195KsVYhEuBK/JhFoYdudd7zf7UV1LQKQfBJKuCnC7Zul8BX8z5y60
T/OQ+jLavjY6OBATydFz99C4Sy7pEg1GtrIjFueYADuZfb+2wfxIDUnstcyJb06e
DZ/pqEkKERhO0AE5+KfK61AVPTPAR31UmDehxUhXvRnzyvaShLNoaOEpeD2Mrwkf
FpqMw8n7t0OA/NEuXiOPYLF0ZrR7R6JB4lTLMf32CoTrL6sEaLLAh5kTQRD1i0wU
Tvq6PruGP1zSApIwAczrOS3Aq8c0+Uzzz1WIRQbN1Mn3EeykIvPRicFovxxTdBOK
kua/yHXUkb94p9KJsOWUUM1Sl1yythPm0bKjlDOdrFmuBcs4EBoVpSUhPY6LPjJl
hLrfiJxuVOuQ8BG87Ar3LaA5MsSgTIqu2qPx3N45mCNakuxwiyf0USvSltc0qnVq
pH4Mp8vsgBFwfXE4b4sPO2/3GKjHVhqUc0yMhWh/Q8vrqGZz7KSilmxXON3zhMzY
rU+0GDxyaWNoQGZsb3djcnlwdC5leGFtcGxlPokCUAQTAQgAOgKeAQKbAQUWAgMB
AAQLCQgHBRUKCQgLFiEEZw9JBGKUIT6hZtumqtVVDfvHnyIFAmVKVvEFCQWyjHAA
CgkQqtVVDfvHnyL1QA//X5hKzJMCzzqFnTZYrTM8iWAa/mNvwn8481XtgeHXrVFX
sGzLoxRRmhuLLYnNKyv6EaAlaZXg5kU49fLQKxi4GPxxRNe/9miWrbqcZdUPwSrt
Fo6Sy02oD7OPktgdkxdSQ90/ZBLcOpepnOCgwqfUI938mWy27drf60SB3TtHfgdZ
tSoCu0Bmw04EtmpexHZX82xHtBVfqSpf5oCJmfv1iaMKoj32V6X8yiXS3t7+yqMZ
Ztan9zbJ+E8+XFIHlGIedDzAt05073Re5jHC+eOwxFnW263r0YfJ52xqGZrN0aH9
PAGg8ZTDHGialG7iZ1gGkanCffoMZS61cXD0Jr58Uc/cRXj04grTfuGcNP+kg7Pk
c13blf2sFsx5Qa8qOnpoGVWHgPRcLK6oaNRcfRoIz4cF+aVhhatjVq/rnBCJ39nV
YNk8dlA7adKBGTifE2pCVTCVHEnT9UGQkulTJmGoW44eU+ro0zbmNkX0kpDFslCp
FRZV8Ptgiaqq83Szgvsh21bPu6EjrXPUYlgbYynrEWlIJ6AzkQE4Ciu5no3Jh2PY
ZY4AWdCG4hMma8IWuECYCfKnzQMGur6CovpDH9+64bHNB8UtuFiwB0RuMul16fSm
2Fryl4Rh4M0ay77lXUztl2Kc9O4KfSse+1l31DHrtflrAJRdiJ2cY68x48Kdr9W0
BFJpY2iJAlAEEwEIADoCngECmwEFFgIDAQAECwkIBwUVCgkICxYhBGcPSQRilCE+
oWbbpqrVVQ37x58iBQJlSlbyBQkFsoxwAAoJEKrVVQ37x58i9j0QAIvu/j6n3omx
UKUcTYZV5BCuUUCy2d7Rw9b3gwQWLlhh/Ag5XDiMj6tYO2JbAQqq3c12urA9GOKP
jmFdPlH4AdRzJhskOo87sjAuj4WgzN8dDFtKl0EEGHlII9KtWzkIYyk8s3bma7H9
T/AOf0PL43DHTn+AjaqbjDftLC28SmpyKVa88HuDw7w0+6fQs003V9KlLBjvDPRB
pWBJU+JGBF1ju1iVLETsZE1eKetjXxShtsc36AAgYyBzghk1ICbnDpgB8KLYrBEs
d3l5Glm00/vD/kLEu/umGgSM3kDubFFicJaWuj78UfkIKtkJcBzjEeFc9JKvl93v
+EklP6YmXqqvWEzOU0ya2ETkYUzwUuessynke2wX7FyMHkVxRLTeEvRGauDJbrp3
svGCvl12JJk0iao8p0snuzgyFmboUAQ3eakMCGY5GrNUi0Va2y8HjXfSh0PbxPXo
GFUp60RrEyKQ6QdIGzn6R9FjUnsjOZN9fX4I867aQp5aLkE5Xn7sPaq5+A8XVr+9
/M45adaf4M9i59h7s2JbiKWnwP9i1QfpYpYAJ+2YYxryF/WpnYJ6BeFY53X8g1uV
47H0oydH9EilfL5PvqwWGtCgzYhDsaJFPZKUP5ABQwFjQvMWwZkoREL4+V9jucuJ
qPhZEbjReUF8tO/gqlUlK5CxVOOMgwLotBtSaWNoIDxyaWNoMUBmbG93Y3J5cHQu
dGVzdD6JAlAEEwEIADoCngECmwEFFgIDAQAECwkIBwUVCgkICxYhBGcPSQRilCE+
oWbbpqrVVQ37x58iBQJlSlbyBQkFsoxwAAoJEKrVVQ37x58ii+IP/A7qrRgXOKYk
SuSKVh2eIi3hPHWU0HH4nLoxiCxhknCmtbaZ8hmv8q8R4PtOoMcp8uyu9sVhmUsS
9p//5yEL8q2h7+C7nCIgL+BFipIFTOIomh3d93NI0+JehP6mtvqfa1dnAo/3rQaa
mPUOvG7G6hRud0PdGYFLYFHO2qtSUdVlvqvrRbcJtLUNXtUbzXzX7x7NXSbgwetf
XihGn3kB61EXEgBAMUJOUOiZ5d12g6Nwa1DKRXbAUFT6gnWSoM39hw/K/zkXt08S
O3OPhLMNN0bn/reG0htky57lFtxjRFvzdQx/673MKhuzaYr5VLU+qO14FZAFNFbf
9lvUZNq/xq59mLTWpLkmu4enpts00Kq2FxUoiMkP1h7ufFXHQRUvu/tZoN9BB5uJ
juXU4BAxMwx144DTaPQVbXaym+RDAQ4JAr4oI3wkzNJm3ajoh1Txvk+9nnbHPUb2
XciiP5WEPWgORLqsa48FJvk7VRJrQ1N7g7R4C3K+f2CbV6O18kQRg4ACW3U3cDL9
wwP/yfC+34vQ82VMNbZFrdEJQ9OkUZJAT46kbseTy3Kpnf9wTAOJvsoXJsMpG+yk
r2E6ZfyyL9W/GfegFx9y+kveFuPXtlOJaTee7wQJTmqnJ/cFmk2L1b7l36SNG502
lUEJ70UBQBuJKdQ5v/5LMFhxr8YgpJUutBtSaWNoIDxyaWNoMkBmbG93Y3J5cHQu
dGVzdD6JAlAEEwEIADoCngECmwEFFgIDAQAECwkIBwUVCgkICxYhBGcPSQRilCE+
oWbbpqrVVQ37x58iBQJlSlbzBQkFsoxwAAoJEKrVVQ37x58iGPcQAII7O4uXF8Q1
rwKYSrgG/qUlqy88MU3Znh16N0bUBdvK5ViYfsepQCDwqF7m8Jc/pqYhs7IZyEdZ
SMz46cjuVM7ebid5/uDm1r0FULB+JAMKm/+mZd/s24Sahs7AMaVgrssrG9PgQjCS
Hr5oCP+BYHmJWIS+TCJPwUlPgOI970QW/tYBiV357VLF5C4RCZala2uaUZVELX3U
JqkzX4dbY2m6Y3g8wUYMZa612AaIWQGRjy2ZN4TgBVf6g/xu5jRf9oqPWuWw7c/C
v8TJkC3S9cpWoo2qTRpe7FHS98XtzykT7CcgF2HSPXpLr3lHkvdrh7GxP8CzWM8E
iTh1kH0hTbvMtqIXTpqR3K/ucdz7xkcOTLAvnWzqs6scY2fCqjnMATArWK7cBcwy
FZ9UA/QL/oKrbbq0nc7ykBFiijnkFiFjSVfQwnDveC7aj3lNu5JCT7nA4x64y5TW
3boB2e311fo5/AcFHXJzBuyW2mgObBQvdDuFNWBf/PR6IHxK+9nDGO2acvfOXroB
rXtnQuBhTgsqkb9mVM8Xu00wEnOSx5EpLQXBsPNKrz5v9QlJH8u9EOUDPUR0D/fu
p/NtRgfbkSgPvNuFyy+Lh88WGNtVWXxH3XsVAQcJBcJxQuRGDr6dLXyrBFP1Ib0s
c7tpQJsxE3M9qoGa03VFvg0+S8LMvqtOtBtSaWNoIDxyaWNoM0BmbG93Y3J5cHQu
dGVzdD6JAlAEEwEIADoCngECmwEFFgIDAQAECwkIBwUVCgkICxYhBGcPSQRilCE+
oWbbpqrVVQ37x58iBQJlSlbzBQkFsoxwAAoJEKrVVQ37x58iIAIQAIysxQBrL/SU
TIEshaiBUcl2hmBxCHZGcQhX/7YSpgdMKBXfiyiKoPB8vJuxcXWEQzBb1PDft3aO
PSwcre0k2P/E9GMRmsSsibtawuS0BmXrxaA4GKB1tInTf8e+uOgkvAgc/vKqt/Ji
rzcqs+/7PQvwZ5HLU0WmiWRWuWrmZM7djPAC505QEWYKOJlmfbX8VQkFbsM5zHcj
YcmiIagCosBwJ7TKJlnuf+NTWJmgPu9UlC1hKyxY7fANQ4jLjyAf9QWQIHa/N7Pl
ThXUfO+tQCxjNtj6Q7xvzNiZS10rs7lFwvo4pSpz+Sj1ikSG0Yh49tcnu7imHCqh
k4G+alk4t0eCkFJ/2nAIwnHZtQCptVfOabPGivA4oclwpL4eOar+PFlfS+3jEIQJ
nlD3zWX0f4GoHS8ntZJpqqu9cG68fqJQy2rwQGciRwa4ifLsExCOZ2KWANOaCzOB
dGvwoMJmsQcgOmpfIXh2x2kxKmS4P7tBpreUI2AgqalOL1910cpByqRvp8eDsAis
mUruBsZaQlrBI3TsfZZEIaxoRtE6HM2mRy96my8GtBc0SBKQ6H9t6eyYBkdcfwTI
JZHRPCjMwWDj3t6vxxKXbAS8HOS4065RFT2pGrpqY0Gp3A+uI5proUAG+wxf9tt/
wJM9SG04GNsI12k7ipSdPvzHEkcGjmszuQENBGU7ZP8BCACawF9hBJzDIKm5OUDU
csawP9EYWK9sWPeII2k35KnnFbncd+dnOwfj6XKw5kMZ02iPtqS8ouaSY0Xxlfm7
lmiCMGDTocMNyQF945+H3vOjA4BAK+sJw8GVg0nThnvycwh/O7ElmfTA+EcZF8ur
OjDJ70og0KLKAfEatFWvBA/VXD7SHcdOAmdpxuty6FVt0IYBFAgSuiNBzBDqTeOD
gPmT1uhaVlI+pqY66UlJM5XaE9jaZuV4DFDcYBmp4pt75AE4erWbO8ZHtMB6E8Qo
3urtf7zMeqeSg2xEES3TX97yfoo5jz0IWotCFgzVRJs6REVQNWEtxljiPIB7HJ6D
VQ/VABEBAAGJAjMEGAEKAB0FAmU7ZP8CngECmyAFFgIDAQAECwkIBwUVCgkICwAK
CRCq1VUN+8efIg4KD/4h5YmCItlVLrLmNCdr1TFl9xhc4pmtmehFpb7SQfuhoatY
XDTVhNN+PRBwf1sy5n7ihAv5XR9S6HvAtWbE9NUnF5D3AT03AIc2LRD8z+GPR0bI
7pg6DueY4BqEBFPV0+KPuYKdUe25uCemhxObfvuu0PSqkuLws/b9h/hS5jyXNqI+
Yuf07AS+kTmjjJqTPXVs3ZvgZ9cZSdsQnTvcCG0bBhWXfAwylSMdpKd1hBiFJ1xY
EJV318TALmmlklIx2fIDQGfj19k0bgVvqhhrIH0i+lo/cc9JfasNghXYKO7QLxnw
fAR307v2/s0dSZUuzJX2vUXbKVIaJpMFIvfBEcrigDq+S0AWsAiZwPtW6M2XSEhv
GxOldMYJsYcsacb3Eldsaf5R0DGO9PTwoAjSZFBBHaxi4QJ6UKFkybD29vuGEK6H
JDhzA3V6cXAsRFR7wIJQRUA1kfmkXf1SlPRAXfLdiLh3WvwfUQY7kXuXoazmzNIY
Bmpn9CWviw5bf1Z8KPZFrGzMFIdSB7dj6uZUaOBbNoGP/X75ttCTbKYgViDTcfHk
YvKL9YG9BBknAoj1p1keLVS0LFVBiEv08K1IR2wj6+zT6QGDf2VuXQAarOy2XSUS
IWy1gQuoQKg2A5NmAlP21n293LoOEjJbL7VSc/ppmjhH58CSzOZkSN2QVqBjM7kB
DQRlO2T/AQgAsPe36RaF6sLH3EDMnxmbOgZwTCwi8leQubMgQYpQ44kxEKTEByci
vO9y7ZlP6eQjGR1MUxnxkPBrbKvZth5wQCDZAqi4X/Pz+m8r0cyuZaZUiegaMfw4
WBLn4iIFWAQ4SXKi24ET7/3qXrgcI4sCVr/YIzdOp0UhMP0V2fXtIbR9XQ0LbYTM
EuDHQc9Riq2RjYafiD9uDgn5VHj5E2qs4rSYWsT4UVNdohX1ELxi7WpwaN3qFf9S
XQMSq1zFgb0HCrxZIB5VD0LlI85rt8A67++hMTcXZ3h4gWutgMrsMeCryKKQpsnt
PMhV8A6qjIa/0b7sZbG6iNkOe+cCVLLnHQARAQABiQNSBBgBCgE8BQJlO2T/Ap4B
ApsCBRYCAwEABAsJCAcFFQoJCAvAXSAEGQEKAAYFAmU7ZP8ACgkQX2WeABQx0RYA
YAf+N6uBrmABFXmGNGTtyirLXGH1znFXZSofijk6AJSOVEznP2pYlXsEAVOlTc7v
kA8nVdXnBjMexWNEZ9L5lyD9U4FM3Sx0kL3GYfuSftKLmDXrseXi7ksPi7uyobL+
Wtn4WA9fp7x4qOj8G/HJO0YdAUI5tSsj+o+Kq7XVJ4IdD7WkWI63NlmIqK/gwFYu
WLmSlhXdg636nhQ5SuxSvRBe8jJUguPbTaaTLo2kaqizgzXDmEk6/zcGR8S42rsb
uvfQnh+wnpdZ5KcnWae0FywFaso0ISxxJG4ljicq1zBwyUxdyU6WwuP1R8mTOABI
Wg7WjDIPkQRIqHgTc+hRHcLZSQAKCRCq1VUN+8efIgb6D/9qXfeLMqiYcVDDUtdT
hFCYzIq5vsSlhSb6VXcXjpxhFt8A9nhWp2GG9xiHWt1gDfEWd3g4KOlm16sEWEPB
6yuZVTfDuzdho3nvavEehry+rTv9tbmpXoqfY8HPacgvXk4I3ZO/MYTnQ2wH1IIP
N3TnEsfSqbwbPgNSh4RrKNRksp26CAZaGepiVZqGCeCjGJtgWy1uIUfhHsplV+9h
ZqgbYr0VdIFWpm4Zkt98ZQcbsH2/MAnqbqWxDXJx1W+Ym6YhagrCt/jsrubUGVN9
9fNdxeQompbYrlrguybK/z0KU0eTFNrqkB9v1LMVxXukNXYlef+xWp6It4JPlRDN
mt/gE4LJQSNqRM/47bRUuUiC/+9Xmwy3vOQcFeWS2z7ij9bav/0y9dNzBxf1WEAA
iFM0pJGaoOM8B7nFyCYrk/gL4acsmYnyregEhLlwzdQsKY/UtjWwJBZfoclIi8GM
VcKCIZJcZXDIKsKuMFRcuaCGfoGqft4gFLj/bfA+jDM7M7aqh7KWThfhTo0LSJKc
7g8aQQ0Q/ZgkpWI3jF7IEooCfkhYYD7bc4K8fNY4oaVO4hmJjihoThduo6KR86Zk
Q2nzB0dIyhDJZFII0BFHDDoFFFWH7YO153EBH3vDudc6NUntLTbxBPqvbyWuUasy
cRJ5dtxyadukcCDDbXPqA4WenbkBDQRlO2T/AQgA6D4Mdi2DFlvpLYnU1mGlYqbh
tVtmjm7QkKQiFx1HCkxLrJLaBIJ7ctNMgcASYKQ4rNYkSPQWWByLScT6EPEQ1zpd
gJ1X2WmPV7On5iYQunZUrl/NXUDJjuxIHRBnJi6tIUqSXf5gBNSBKp0kMA3qBySl
zKtqI7Wjib3YIh3IhQu+BIWhbE0KU22TnlomKZd3CpTx2BJEROoSlDnMFrGYKWWF
jiwsphaMNOJVhkbGIPxcolD9PmK1PPmicJX5OibCRg8XX29y7qYB+Vp0Z+EYeVXz
sw1b+lW9piXcpkvLQwL5sbbgxIar5jl3RgeK6KmrxybzyKKmLNvHfiYyqbx4sQAR
AQABiQJQBBgBCAA6Ap4BApsMBRYCAwEABAsJCAcFFQoJCAsWIQRnD0kEYpQhPqFm
26aq1VUN+8efIgUCZUpTHQUJCXTvngAKCRCq1VUN+8efIoleD/4nfu7ib+NJO3BN
eDgf3IaZh9TDjp748EoTtvZxDjdEWivr+7zEZdFHCGMzOqcMQmMcqz6twfVY05tI
Z++SF6OWEAP5FJTpnLn+UfCxwsG1MwFATFQi3fB4V+1pfBQoZ0fZYbZMr/xAisKG
SJ6Y1TziaFrfQw3O7yPbLbKCBhwTSeKDu24TTypEezctZNvduMjckc3FEBrlOlRR
/lJmho6AN7uBM+1P89bfpDQuPzxXO90BKqliInV6XVCsLAbeIaYgNeuACe6/hbqh
P6sMoi5ifB9XZw9YEPTm/a8NG54drNh0t2U0PK+I9n1+KIs/d3GjwASSM+eI9awp
biVT7bpxfi9avhMI1eX1rivuXTM+R3KNxKyKVhP61Fpm8s3shkJx/LCKpcAQedrH
PuOosEAzLBXyPs24ela4Xzwiqlf+4waMxSzXRTQPXGvgpwvs1WmGhC8ilOQEXhiB
EakEjo8HA4SyFDtpKLxetSAr/N/R0biTa7AYXYHEi3Vp2tnQ5RxvGuUJAGxJNw0J
1wNdC/QRBxCzrId1sBMZ1Zuenh/jS75ZJhNpBw8ddksxF+UbuChoIoUqFGS/DyKu
WDldLAjOW1tlgF+bMArgq1wQ49uiVQW5F4JGsx1xubMv3rU/q3mnbOuAPVUg+u29
6ZWH+6uRa2WOwTOixrHAvc7C3n0guw==
=wwOF
-----END PGP PUBLIC KEY BLOCK-----
```

</details>

<details>
  <summary>Key 2</summary>

```
-----BEGIN PGP PUBLIC KEY BLOCK-----
Version: FlowCrypt 1.4.7_dev_147__2023_11_09__3227f2d Gmail Encryption
Comment: Seamlessly send and receive encrypted email

mDMEZTtl2hYJKwYBBAHaRw8BAQdAETjbimsYHDnqAcoif3zn7fwzm5PMPGW3tA2v
4o7Sdom0GlJpY2ggPHJpY2hAZmxvd2NyeXB0LnRlc3Q+iI8EExYKAEEFAmU7ZdoJ
EMa0/MqQUOp1FiEEN3iaVHzSUOmW++DGxrT8ypBQ6nUCngECmwEFFgIDAQAECwkI
BwUVCgkICwKZAQAAJJ8BAIpmnmy3EhHyXf6Etx6vJuB20RVvrBq7UR+6Twfg6z56
AQCB9/9UTSn6D+XyCJrZ0U9iGAJnVBYe7J4Vz9wNakaKBLQdUmljaCA8cmljaEBm
bG93Y3J5cHQuZXhhbXBsZT6IjAQTFgoAPgUCZTtl2gkQxrT8ypBQ6nUWIQQ3eJpU
fNJQ6Zb74MbGtPzKkFDqdQKeAQKbAQUWAgMBAAQLCQgHBRUKCQgLAADpeAEAsFou
G2qNz5AhCvRV2T7EJJ9GfDnt3AAKX8q9BXEX9V4A/0VmlGqKAm2qzPpTjQHkLh/Q
VHVy4t+xK3SBfeLcy4IDuDMEZTtl2hYJKwYBBAHaRw8BAQdA4BUl1QPc7/Fcmkao
iMiiXDsvFniFRra7FdQGrtSnKkGI1QQYFgoAfQUCZTtl2gKeAQKbAgUWAgMBAAQL
CQgHBRUKCQgLXyAEGRYKAAYFAmU7ZdoACgkQ88L/I3CCI5RLnwEA/uCQEgzzqTuK
LbbBimczL40rEtcA9DMktTIxkBkqoK8A+waYoqHx7jhqGNAB3Ut4w07GbEoxWMjN
0TNkT9U6uREGAAoJEMa0/MqQUOp18GsA/Rr9OGhzZsmxNwvxS/jBhnbbhn34+P5U
+UwqXonudU5/AQDybdpEt6EwyDtsKT6wu8QP0f+bNh3rzfdRhOTQYePhAA==
=/TSG
-----END PGP PUBLIC KEY BLOCK-----

```
</details>

### PR DESCRIPTION
**Pull request by martgil**
This PR adds a check for an incoming public key to determine the correct statement for the import button, depending on whether it's already in the contacts.
close #5480 

----------------------------------

**Tests** _(delete all except exactly one)_:
- Tests added or updated

--------------------------------

### To be filled by reviewers

I have reviewed that this PR... _(tick whichever items you personally focused on during this review)_:
- [x] addresses the issue it closes (if any)
- [x] code is readable and understandable
- [x] is accompanied with tests, or tests are not needed
- [x] is free of vulnerabilities
- [x] is documented clearly and usefully, or doesn't need documentation

### CODE REVIEW

**Review by sosnovsky — COMMENTED** [review]
Works well, just a small code improvements

File: extension/css/component.css [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+/* Badges */
+.fc-badge {
+  color: #eee;
+  font-size: 0.7em;
+  font-weight: 600;
+  text-transform: uppercase;
+  border-radius: 3px;
+  display: inline-block;
+  padding: 0.2em 0.35em;
+}
```
[the remark this answers is not in the document]
Comment (martgil) [reply extension/css/component.css:12]: will update it. thanks!

File: extension/css/cryptup.css [hunk 2]
```diff
@@ -630,22 +630,32 @@ body#settings > div#content.contacts #emails .action_show_pubkey_list.opened .ic
...
 }
 
+body#settings > div#content.contacts #emails .contacts-pubkey .contacts-pubkey-info {
+  display: inline-block;
+}
+
+body#settings > div#content.contacts #emails .contacts-pubkey .contacts-pubkey-actions {
+  display: inline-block;
+}
+
```
[the remark this answers is not in the document]
Comment (martgil) [reply extension/css/cryptup.css]: thanks, will do.

File: extension/css/component.css [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+.fc-badge-gray {
+  background-color: rgb(97, 97, 97);
+}
+
+.fc-badge-light-gray {
+  background-color: rgb(188, 188, 188);
+  color: #444;
+}
```
[the remark this answers is not in the document]
Comment (martgil) [reply extension/css/component.css]: my bad, ill use the hex values.

File: extension/chrome/elements/pgp_pubkey.ts [hunk 4]
```diff
@@ -158,7 +158,13 @@ View.run(
...
         $('.action_add_contact')
           .text(
-            contactWithPubKeys && contactWithPubKeys.sortedPubkeys && contactWithPubKeys.sortedPubkeys.length > 0
+            contactWithPubKeys &&
+              contactWithPubKeys.sortedPubkeys &&
+              contactWithPubKeys.sortedPubkeys.length > 0 &&
+              this.parsedPublicKeys &&
+              contactWithPubKeys.sortedPubkeys.some(
+                existing => this.parsedPublicKeys && this.parsedPublicKeys.some(parsedPubkey => existing.pubkey.id === parsedPubkey.id)
+              )
```
[the remark this answers is not in the document]
Comment (martgil) [reply extension/chrome/elements/pgp_pubkey.ts]: thanks, i will ensure that complex-looking process to be descriptive next time.

**Comment by martgil** [comment]
@sosnovsky I have satisfied all the requested changes. please let me know if it looks good to you. thank you !

File: test/source/tests/page-recipe/settings-page-recipe.ts [hunk 5]
```diff
@@ -168,4 +168,15 @@ export class SettingsPageRecipe extends PageRecipe {
...
+
+  public static checkContactKey = async (
+    contactsFrame: ControllableFrame,
+    keyType: 'OPENPGP' | 'X509',
+    keyStatus: 'ACTIVE' | 'SIGN ONLY' | 'EXPIRED' | 'UNUSABLE' | 'REVOKED',
+    fingerprint: string
+  ) => {
+    await contactsFrame.waitForContent('@page-contacts', fingerprint);
+    await contactsFrame.waitForContent('@container-contact-key-type', keyType);
+    await contactsFrame.waitForContent('@container-key-status', keyStatus);
```
Comment (sosnovsky) [inline test/source/tests/page-recipe/settings-page-recipe.ts:180]: it seems we can get incorrect test results when there are several keys on contacts screen. as when checking `await contactsFrame.waitForContent('@container-contact-key-type', keyType);` we don't explicitly specify which key we're checking.

the same for `await contactsFrame.waitForContent('@container-key-status', keyStatus);`

to fix this we need to explicitly specify key id in `data-test` property (like it's currently done for `action-show-pubkey-${keyid}`

then we'll be able to check status and type for correct key, like this `await contactsFrame.waitForContent('@container-key-status-${keyId}', keyStatus);`

**Review by sosnovsky — COMMENTED** [review]
Looks good, just a couple small fixes left

File: extension/chrome/elements/pgp_pubkey.ts [hunk 6]
```diff
@@ -156,12 +156,16 @@ View.run(
...
       } else {
         const contactWithPubKeys = await ContactStore.getOneWithAllPubkeys(undefined, String($('.input_email').val()));
+        const isExistingKey =
+          contactWithPubKeys &&
+          contactWithPubKeys.sortedPubkeys &&
+          contactWithPubKeys.sortedPubkeys.length > 0 &&
+          this.parsedPublicKeys &&
+          contactWithPubKeys.sortedPubkeys.some(
+            existing => this.parsedPublicKeys && this.parsedPublicKeys.some(parsedPubkey => existing.pubkey.id === parsedPubkey.id)
+          );
```
[the remark this answers is not in the document]
Comment (martgil) [reply extension/chrome/elements/pgp_pubkey.ts:166]: Will do, thanks for the remarks.

**Comment by martgil** [comment]
@sosnovsky Does it look good now?

**Pull request merged**